polyglot/piranha/src/cleanup_rules/go/rules.toml

```diff
@@ -0,0 +1,487 @@
+#   nil == nil     -> true
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_identity_equal"
+query = """
+(
+    (binary_expression
+        left: (_) @lhs
+        operator: "=="
+        right: (_) @rhs
+    ) @binary_expression
+    (#eq? @lhs @rhs)
+)
+"""
+replace = "true"
+replace_node = "binary_expression"
+
+# Simplifies not equal identity comparison
+# Note that `nil != nil` is not compilable in Go, but compiles in tree-sitter
+#   true != true   -> false
+#   false != false -> false
+#   nil != nil     -> false
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_identity_not_equal"
+query = """
+(
+    (binary_expression
+        left: (_) @lhs
+        operator: "!="
+        right: (_) @rhs
+    ) @binary_expression
+    (#eq? @lhs @rhs)
+)
+"""
+replace = "false"
+replace_node = "binary_expression"
+
+# Dummy rule that acts as a junction for all statement based cleanups
+[[rules]]
+name = "statement_cleanup"
+
+# Before :
+#  if (true) { doSomething(); }
+# After :
+#  { doSomething(); }
+#
+# Before :
+#  if (true) { doSomething(); } else { doSomethingElse();}
+# After :
+#  { doSomething(); }
+#
+[[rules]]
+groups = ["if_cleanup"]
+name = "simplify_if_statement_true"
+query = """
+(
+    (if_statement
+        condition : (
+            [
+                (true)
+                (parenthesized_expression (true))
+            ]
+        )
+        consequence : ((block) @consequence)
+    ) @if_statement
+)
+"""
+replace = "@consequence"
+replace_node = "if_statement"
+
+# Before :
+#  if (true) { doSomething(); } else { doSomethingElse();}
+# After :
+#  { doSomethingElse(); }
+#
+# Before :
+#  if (true) { doSomething(); }
+# After :
+#
+[[rules]]
+groups = ["if_cleanup"]
+name = "simplify_if_statement_false"
+query = """
+(
+    (if_statement
+        condition : (
+            [
+                (false)
+                (parenthesized_expression (false))
+            ]
+        )
+        consequence : (_)
+        alternative: ((_) @alternative) ?
+    ) @if_statement
+)
+"""
+replace = "@alternative"
+replace_node = "if_statement"
+
+# Before :
+#  {
+#     someStepsBefore();
+#     {
+#        someSteps();
+#     }
+#     someStepsAfter();
+#  }
+# After :
+#  {
+#     someStepsBefore();
+#        someSteps();
+#     someStepsAfter();
+#  }
+#
+# Note that we need to tag basically all nodes here.
+# Including not so obvious ones: @outer.stmt_list and @outer.block
+[[rules]]
+name = "remove_unnecessary_nested_block"
+query = """
+(
+    (block
+        (statement_list
+            (((_) "\\n"?)* @pre)
+            ((block
+                (statement_list) @nested.statements
+            ) @nested.block)
+            (((_) "\\n"?)* @post)
+        ) @outer.stmt_list
+    ) @outer.block
+)
+"""
+replace = "@nested.statements"
+replace_node = "nested.block"
+
+#####
+# Dummy rule to introduce a cycle for `delete_statement_after_return`
+[[rules]]
+name = "return_cleanup"
```

Comment:
I think there is some naming convention for dummy rules in the readme

Reply:
Renamed to `return_statement_cleanup`